lib/mauth/rack.rb

```diff
@@ -70,12 +86,23 @@ def should_authenticate?(env)
       end
     end
 
-    # signs outgoing responses
+    # signs outgoing responses with only the protocol used to sign the request.
     class ResponseSigner < MAuth::Middleware
       def call(env)
         unsigned_response = @app.call(env)
-        signed_response = mauth_client.signed(MAuth::Rack::Response.new(*unsigned_response))
-        signed_response.status_headers_body
+
+        method =
+          if env['mauth.protocol_version'] == 2
+            :signed_v2
+          elsif env['mauth.protocol_version'] == 1
+            :signed_v1
+          else
+            # if no protocol was supplied then use `signed` which either signs
+            # with both protocol versions or only v2
+            :signed
+          end
```

Comment:
is this a case that actually occurs?  like if we get a request with no mauth header whatsoever, we are still signing the error response back to them that says missing header?

Reply:
Was actually just testing that out and now we don't sign the "Unauthorized" responses so I'll take out this else.

Reply:
@ejinotti-mdsol fyi added this back in because this case can be hit:
One example i ran into while local testing the new client with eureka: eureka does not authenticate requests locally by default but still signs them.